source/Nevermore/IQueryExecutor.cs

```diff
@@ -76,6 +76,14 @@ public interface IQueryExecutor
         /// <returns>A stream of resulting documents.</returns>
         ITableSourceQueryBuilder<TDocument> TableQuery<TDocument>() where TDocument : class, IId;
 
+        /// <summary>
+        /// Returns strongly typed documents from the specified raw SQL query.
+        /// </summary>
+        /// <typeparam name="TDocument">The type of document being queried. Results from the database will be mapped to this type.</typeparam>
+        /// <param name="query">The SQL query to execute. Example: <c>SELECT COUNT(*) FROM...</c></param>
+        /// <returns>A builder to further customize the query.</returns>
+        ISubquerySourceBuilder<TDocument> RawSqlQuery<TDocument>(string query) where TDocument : class, IId;
```

Comment:
I'd love to remove the constraint that `TDocument : IId`. In the integration test, we were forced to set a dummy `Id` on `ProductSubtotal`, but it's not used. I'm having a look at whether we can remove this constraint; input welcome.

Reply:
Yeh it doesn't make sense in this context

Reply:
Thanks @droyad. The `IId` constraint was there because we add [an `ORDER BY [Id]` clause](https://github.com/OctopusDeploy/Nevermore/blob/master/source/Nevermore/SelectBuilder.cs#L177) by default in some cases.

I took a stab at removing the constraint along with the default `ORDER BY` clause in [this commit](https://github.com/OctopusDeploy/Nevermore/pull/69/commits/4eb4a2626725a5302016df7c2ec4468a2c85c85d), but I'll revert it for now because there are some locations in Octopus server where [we seem to rely](https://github.com/OctopusDeploy/OctopusDeploy/blob/master/source/Octopus.Server/Web/Api/Actions/EnvironmentsMachinesResponder.cs#L86) on this default ordering behavior. Too risky for too little gain for me right now.

Reply:
Ah yeh, we definitely need the default ordering on most queries (for paging and other things as you have found).